src/pages/PlaceAddModal.tsx

```diff
@@ -0,0 +1,196 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+
+import { CalendarType } from '@/constants/CalendarType';
+import { getPlaceById, useAddPlaceMutation } from '@/hooks/api/places';
+import { useGetTypesByScopeQuery } from '@/hooks/api/types';
+import { useHeaders } from '@/hooks/api/useHeaders';
+import { Button, ButtonVariants } from '@/ui/Button';
+import { FormElement } from '@/ui/FormElement';
+import { Inline } from '@/ui/Inline';
+import { Input } from '@/ui/Input';
+import { Modal, ModalSizes, ModalVariants } from '@/ui/Modal';
+import { Paragraph } from '@/ui/Paragraph';
+import { Stack } from '@/ui/Stack';
+import { Text, TextVariants } from '@/ui/Text';
+
+import { City } from './CityPicker';
+
+const schema = yup
+  .object({
+    name: yup.string().required(),
+    streetAndNumber: yup.string().required(),
+    zip: yup.string().required(),
+    municipalityName: yup.string().required(),
+    type: yup.object({
+      label: yup.string().required(),
+      domain: yup.string().required(),
+      id: yup.string().required(),
+    }),
+  })
+  .required();
+
+type FormData = yup.InferType<typeof schema>;
+
+type Props = {
+  visible: boolean;
+  municipality: City;
+  prefillPlaceName: string;
+  onClose: () => void;
+  onConfirmSuccess: (place: any) => void;
+};
+
+const PlaceAddModal = ({
+  visible,
+  onClose,
+  municipality,
+  prefillPlaceName,
+  onConfirmSuccess,
+}: Props) => {
+  const { t, i18n } = useTranslation();
+
+  const getTypesByScopeQuery = useGetTypesByScopeQuery({
+    scope: 'places',
+  });
+
+  const headers = useHeaders();
+
+  const types = getTypesByScopeQuery.data ?? [];
+
+  const addPlaceMutation = useAddPlaceMutation();
+
+  const handleConfirm = async () => {
+    await handleSubmit(async (data) => {
+      const formData = {
+        address: {
+          addressCountry: 'BE', // TODO get country from event form
+          addressLocality: data.municipalityName,
+          postalCode: data.zip,
+          streetAddress: data.streetAndNumber,
+        },
+        calendar: {
+          calendarType: CalendarType.PERMANENT,
+        },
+        mainLanguage: i18n.language,
+        name: data.name,
+        type: data.type,
+      };
+
+      const resp = await addPlaceMutation.mutateAsync({ ...formData });
+
+      if (resp.placeId) {
+        const newPlace = await getPlaceById({ headers, id: resp.placeId });
+        onConfirmSuccess(newPlace);
+      }
```

Comment:
```suggestion
      if (!resp?.placeId) return;
      
      const newPlace = await getPlaceById({ headers, id: resp.placeId });
      onConfirmSuccess(newPlace);
```

You could write it like this